src/PhpSpec/Runner/Maintainer/CollaboratorsMaintainer.php

```diff
@@ -140,6 +130,9 @@ private function generateCollaborators(CollaboratorManager $collaborators, \Refl
                 if (($indexedClass = $this->getParameterTypeFromIndex($classRefl, $parameter))
                     || ($indexedClass = $this->getParameterTypeFromReflection($parameter))) {
                     $collaborator->beADoubleOf($indexedClass);
+                    if (null !== $indexedClass) {
```

Comment:
Oh, right. PHP treats a null assignement as false, right? So, this check is just obsolete. If removed, it will work just as intended.